### ISSUE TITLE
`ns-render dataset` bug for gaussian splatting

### ISSUE THREAD

**Issue by pa-la** [nerfstudio-project/nerfstudio#2913]
I want to render a splatfacto model using the command line
```
ns-render dataset --load-config=/path/to/config.yml --output-path=/path/to/render_dir --split=train+test --rendered_output_names=rgb
```

I obtain the following error:

```
Traceback (most recent call last):
  File "/home/ubuntu/contribs/nerfstudio/nerfstudio/scripts/render.py", line 898, in <module>
    entrypoint()
  File "/home/ubuntu/contribs/nerfstudio/nerfstudio/scripts/render.py", line 894, in entrypoint
    tyro.cli(Commands).main()
  File "/home/ubuntu/contribs/nerfstudio/nerfstudio/scripts/render.py", line 733, in main
    config, pipeline, _, _ = eval_setup(
  File "/home/ubuntu/contribs/nerfstudio/nerfstudio/utils/eval_utils.py", line 97, in eval_setup
    config = update_config_callback(config)
  File "/home/ubuntu/contribs/nerfstudio/nerfstudio/scripts/render.py", line 721, in update_config
    assert isinstance(data_manager_config, VanillaDataManagerConfig)
AssertionError
```

However, when I comment https://github.com/nerfstudio-project/nerfstudio/blob/7f9def1ff0b869dcf1d00abe475759de05cdf435/nerfstudio/scripts/render.py#L721 and https://github.com/nerfstudio-project/nerfstudio/blob/7f9def1ff0b869dcf1d00abe475759de05cdf435/nerfstudio/scripts/render.py#L740 I obtain the expected result. 

Are these assertions absolutely necessary? Should they be updated to also accept the `FullImageDatamangerConfig` that is used by splatfacto?

I'm happy to raise a PR once this point is clarified.

**Comment by kerrj** [comment]
Good catch! yes they should also include `FullImageDatamangerConfig`, I think it might have just fallen through the cracks since we don't use the dataset render often

**Comment by pa-la** [comment]
Thanks for your feedback, it's indeed a very useful feature!
I created the following PR to fix that: https://github.com/nerfstudio-project/nerfstudio/pull/2917

**Comment by ataraxyz** [comment]
Hey, this feature is valuable indeed. My issue is that the cameras dont match up after their positions get optimized during training. If those flags are turned OFF, the cameras match perfectly. Can I request the already-optimized cameras? So that even after training the cameras match?
render.py should be where this is at?

**Comment by mzlchou** [comment]
> Thanks for your feedback, it's indeed a very useful feature! I created the following PR to fix that: #2917

It seems like the problem got fixed, but I still got an error and only got it to work by commenting out the lines. 

assert isinstance(data_manager_config, (VanillaDataManagerConfig, FullImageDatamanagerConfig))
AssertionError
